core/src/test/java/org/apache/iceberg/TestBaseIncrementalAppendScan.java

```diff
@@ -50,6 +51,131 @@ public void testFromSnapshotInclusive() {
     Assert.assertEquals(3, Iterables.size(scanWithToSnapshot.planFiles()));
   }
 
+  @Test
+  public void testFromSnapshotInclusiveWithRef() {
+    table.newFastAppend().appendFile(FILE_A).commit();
+    long snapshotAId = table.currentSnapshot().snapshotId();
+
+    String branchName = "b1";
+    String tagSnapshotAName = "t1";
+    table.manageSnapshots().createBranch(branchName, snapshotAId).commit();
+    table.manageSnapshots().createTag(tagSnapshotAName, snapshotAId).commit();
+
+    String tagSnapshotBName = "t2";
+    table.newFastAppend().appendFile(FILE_B).appendFile(FILE_B).commit();
+    long snapshotBId = table.currentSnapshot().snapshotId();
+    table.manageSnapshots().createTag(tagSnapshotBName, snapshotBId).commit();
+    table.newFastAppend().appendFile(FILE_C).appendFile(FILE_C).commit();
+    long snapshotCId = table.currentSnapshot().snapshotId();
+
+    IncrementalAppendScan scan = newScan().fromSnapshotInclusive(tagSnapshotAName);
+    Assert.assertEquals(5, Iterables.size(scan.planFiles()));
```

Comment:
```suggestion
    Assertions.assertThat(scan.planFiles()).hasSize(5);
```

Reply:
this is typically preferrable because it will print you the content in case the assertion ever fails (see below example where I changed the expected size to 4):

```
Expected size: 4 but was: 5 in:
[BaseFileScanTask{file=/path/to/data-b.parquet, partition_data=PartitionData{data_bucket=1}, residual=true},
    BaseFileScanTask{file=/path/to/data-a.parquet, partition_data=PartitionData{data_bucket=0}, residual=true},
    BaseFileScanTask{file=/path/to/data-c.parquet, partition_data=PartitionData{data_bucket=2}, residual=true},
    BaseFileScanTask{file=/path/to/data-b.parquet, partition_data=PartitionData{data_bucket=1}, residual=true},
    BaseFileScanTask{file=/path/to/data-c.parquet, partition_data=PartitionData{data_bucket=2}, residual=true}]
```

Would be good to apply this across the newly added tests.

Reply:
Good suggestion.